include/irr/asset/filters/dithering/CPrecomputedDither.h

```diff
@@ -52,12 +52,12 @@ namespace irr
 
 						struct
 						{
-							asset::ICPUBuffer* buffer;
+							const asset::ICPUBuffer* buffer;
 							core::vectorSIMDu32 strides;
 							asset::E_FORMAT format;
 						} ditherImageData;
 
-						core::smart_refctd_ptr<asset::ICPUImage> ditheringImage;
+						const asset::ICPUImage* const ditheringImage;
```

Comment:
you don't need to keep it around after you construct the state.

Reply:
Look at how stare gets constructed, I think its okay

Reply:
state

Reply:
you dont want to keep the image around (regions, etc.) just its contents.

Reply:
corrected in following commit